src/plugins/workspace/public/utils.ts

```diff
@@ -29,3 +29,53 @@ export const formatUrlWithWorkspaceId = (
 
   return newUrl.toString();
 };
+
+/**
+ * Given a list of feature config, check if a feature matches config
+ * Rules:
+ * 1. `*` matches any feature
+ * 2. config starts with `@` matches category, for example, @management matches any feature of `management` category
+ * 3. to match a specific feature, just use the feature id, such as `discover`
+ * 4. to exclude feature or category, use `!@management` or `!discover`
+ * 5. the order of featureConfig array matters, from left to right, the later config override the previous config,
+ * for example, ['!@management', '*'] matches any feature because '*' overrides the previous setting: '!@management'
+ */
+export const featureMatchesConfig = (featureConfigs: string[]) => ({
+  id,
+  category,
+}: {
+  id: string;
+  category?: AppCategory;
+}) => {
+  let matched = false;
+
+  for (const featureConfig of featureConfigs) {
+    // '*' matches any feature
+    if (featureConfig === '*') {
+      matched = true;
+    }
+
+    // The config starts with `@` matches a category
+    if (category && featureConfig === `@${category.id}`) {
+      matched = true;
+    }
+
+    // The config matches a feature id
+    if (featureConfig === id) {
+      matched = true;
+    }
+
+    // If a config starts with `!`, such feature or category will be excluded
+    if (featureConfig.startsWith('!')) {
```

Comment:
For featureConfigs that only have one config like "!dev_tools",  the result would be `false` when validate `match({ id: 'integrations', category: { id: 'management', label: 'Management' } })`. Is this by intention?

Reply:
Yes, my thinking was, if user wants to match everything but not `dev_tools`, then the config should look like this: `['*', '!dev_tools']`